python/image_adapter/test_train_numpy_smoke.py

```
@@ -0,0 +1,85 @@
+                    "target_tokens": [i % 8, (i + 1) % 8, (i + 2) % 8, (i + 3) % 8],
+                    "codebook_size": 8,
+                }
+                for i in range(4)
+            ]
+            data_path.write_text(
+                "".join(json.dumps(row, sort_keys=True) + "\n" for row in rows),
+                encoding="utf-8",
+            )
+
+            script = Path(__file__).with_name("train_numpy.py")
+            subprocess.run(
```

Ruff (0.15.14)

[error] `subprocess` call: check for execution of untrusted input

(S603)